navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.dualscreen.navigationrail
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.graphics.Color
+import android.graphics.Point
+import android.graphics.Rect
+import android.graphics.drawable.ColorDrawable
+import android.os.Build
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.MotionEvent
+import android.view.View
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.view.animation.BaseInterpolator
+import androidx.activity.ComponentActivity
+import androidx.core.view.animation.PathInterpolatorCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.transition.ChangeBounds
+import androidx.transition.Transition
+import androidx.transition.TransitionManager
+import androidx.window.layout.WindowInfoTracker
+import androidx.window.layout.WindowLayoutInfo
+import com.google.android.material.navigationrail.NavigationRailMenuView
+import com.google.android.material.navigationrail.NavigationRailView
+import com.microsoft.device.dualscreen.utils.wm.DisplayPosition
+import com.microsoft.device.dualscreen.utils.wm.OnVerticalSwipeListener
+import com.microsoft.device.dualscreen.utils.wm.ScreenMode
+import com.microsoft.device.dualscreen.utils.wm.extractFoldingFeatureRect
+import com.microsoft.device.dualscreen.utils.wm.getWindowVisibleDisplayFrame
+import com.microsoft.device.dualscreen.utils.wm.isFoldingFeatureHorizontal
+import com.microsoft.device.dualscreen.utils.wm.isInDualMode
+import com.microsoft.device.dualscreen.utils.wm.locationOnScreen
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+const val MARGIN_LOWERING_FACTOR = 0.95
+
+/**
+ * A sub class of the [NavigationRailView] that can position its children in different ways when the application is spanned on both screens.
+ * Using the [arrangeButtons] and [setMenuGravity] the children can be split in any way between the two screens.
+ * Animations can be used when changing the arrangement of the buttons on the two screen.
+ */
+class NavigationRailView : NavigationRailView {
+    constructor(context: Context) : super(context) {
+        this.registerWindowInfoFlow()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        this.registerWindowInfoFlow()
+        extractAttributes(context, attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        this.registerWindowInfoFlow()
+        extractAttributes(context, attrs)
+        menuGravity = Gravity.TOP
```

Comment:
This line of code should be only inside this constructor?

Reply:
Good catch, it is useless because Gravity.TOP is the default